lastgersync/processor.go

```diff
@@ -2,292 +2,135 @@ package lastgersync
-		Path(dbPath).
-		WithTableCfg(tableCfgFunc).
-		Open()
+	db, err := db.NewSQLiteDB(dbPath)
 	if err != nil {
 		return nil, err
 	}
-
+	logger := log.WithFields("bridge-syncer", loggerPrefix)
 	return &processor{
-		db: db,
+		db:  db,
+		log: logger,
 	}, nil
 }
 
-// GetLastProcessedBlockAndL1InfoTreeIndex returns the last processed block oby the processor, including blocks
+// GetLastProcessedBlock returns the last processed block by the processor, including blocks
 // that don't have events
 func (p *processor) GetLastProcessedBlock(ctx context.Context) (uint64, error) {
-	tx, err := p.db.BeginRo(ctx)
-	if err != nil {
-		return 0, err
-	}
-	defer tx.Rollback()
-
-	return p.getLastProcessedBlockWithTx(tx)
-}
-
-func (p *processor) getLastIndex(ctx context.Context) (uint32, error) {
-	tx, err := p.db.BeginRo(ctx)
-	if err != nil {
-		return 0, err
-	}
-	defer tx.Rollback()
-
-	return p.getLastIndexWithTx(tx)
-}
-
-func (p *processor) getLastIndexWithTx(tx kv.Tx) (uint32, error) {
-	iter, err := tx.RangeDescend(gerTable, common.Uint32ToBytes(math.MaxUint32), common.Uint32ToBytes(0), 1)
-	if err != nil {
-		return 0, err
-	}
-	k, _, err := iter.Next()
-	if err != nil {
-		return 0, err
-	}
-	if k == nil {
-		return 0, db.ErrNotFound
+	var lastProcessedBlock uint64
+	row := p.db.QueryRow("SELECT num FROM BLOCK ORDER BY num DESC LIMIT 1;")
+	err := row.Scan(&lastProcessedBlock)
+	if errors.Is(err, sql.ErrNoRows) {
+		return 0, nil
 	}
-
-	return common.BytesToUint32(k), nil
+	return lastProcessedBlock, err
 }
 
-func (p *processor) getLastProcessedBlockWithTx(tx kv.Tx) (uint64, error) {
-	if lastProcessedBytes, err := tx.GetOne(lastProcessedTable, lastProcessedKey); err != nil {
-		return 0, err
-	} else if lastProcessedBytes == nil {
+func (p *processor) getLastIndex() (uint32, error) {
+	var lastIndex uint32
+	row := p.db.QueryRow(`
+		SELECT l1_info_tree_index 
+		FROM global_exit_root 
+		ORDER BY l1_info_tree_index DESC LIMIT 1;
+	`)
+	err := row.Scan(&lastIndex)
+	if errors.Is(err, sql.ErrNoRows) {
 		return 0, nil
-	} else {
-		return common.BytesToUint64(lastProcessedBytes), nil
 	}
-}
-
-func (p *processor) updateLastProcessedBlockWithTx(tx kv.RwTx, blockNum uint64) error {
-	return tx.Put(lastProcessedTable, lastProcessedKey, common.Uint64ToBytes(blockNum))
+	return lastIndex, err
 }
 
 func (p *processor) ProcessBlock(ctx context.Context, block sync.Block) error {
-	tx, err := p.db.BeginRw(ctx)
+	tx, err := db.NewTx(ctx, p.db)
 	if err != nil {
 		return err
 	}
-
-	lenEvents := len(block.Events)
-	var lastIndex int64
-	if lenEvents > 0 {
-		li, err := p.getLastIndexWithTx(tx)
-		switch {
-		case errors.Is(err, db.ErrNotFound):
-			lastIndex = -1
-
-		case err != nil:
-			tx.Rollback()
-			return err
-
-		default:
-			lastIndex = int64(li)
+	shouldRollback := true
+	defer func() {
+		if shouldRollback {
+			if errRllbck := tx.Rollback(); errRllbck != nil {
+				log.Errorf("error while rolling back tx %v", errRllbck)
```

Comment:
Nitpick, only 2 letters longer but nicer IMO, if you agree (not pushing, just a suggestion though). :slightly_smiling_face:
```suggestion
			if errRollback := tx.Rollback(); errRollback != nil {
				log.Errorf("error while rolling back tx %v", errRollback)
```